torchtext/experimental/transforms.py

```diff
@@ -144,22 +147,169 @@ class TextSequentialTransforms(nn.Sequential):
 
         Example:
             >>> import torch
-            >>> from torchtext.experimental.transforms import BasicEnglishNormalize, TextSequentialTransforms
-            >>> tokenizer = BasicEnglishNormalize()
-            >>> txt_pipeline = TextSequentialTransforms(tokenizer)
+            >>> from torchtext.experimental.transforms import PretrainedSPTransform, ToLongTensor, TextSequentialTransforms
+            >>> spm_transform = PretrainedSPTransform()
+            >>> to_tensor = ToLongTensor()
+            >>> txt_pipeline = TextSequentialTransforms(spm_transform, to_tensor)
             >>> jit_txt_pipeline = torch.jit.script(txt_pipeline)
     """
-    def forward(self, input: str):
+    def forward(self, line: str):
+        r"""
+        Args:
+            line: the input string
+
+        Examples:
+            >>> txt_pipeline('the pretrained sp model names')
+        """
         for module in self:
-            input = module(input)
-        return input
+            line = module(line)
+        return line
 
     def to_ivalue(self):
-        r"""Return a JITable TextSequentialTransforms.
+        r"""Call to_ivalue func if any modules have the func.
         """
         module_list = []
         for _idx, _module in enumerate(self):
             if hasattr(_module, 'to_ivalue'):
                 _module = _module.to_ivalue()
             module_list.append((str(_idx), _module))
         return TextSequentialTransforms(OrderedDict(module_list))
+
+
+_pretrained_spm = ['text_unigram_15000', 'text_unigram_25000', 'text_unigram_50000',
+                   'text_bpe_15000', 'text_bpe_25000', 'text_bpe_50000']
+
+
+class PretrainedSPTokenizer(nn.Module):
```

Comment:
Why is this necessary instead of using torch.load on a JIT'd instance of our sentencepiece class?

Reply:
Or I guess, why have a wrapper class like this at all instead of just a list of URLs?

Reply:
This is a standard transform that can be used together with other transform. Users can download and load a spm but still need a wrapper (cannot jit a method of JITable class).  User can also use this transform for any local pretrained spm. For example, pytext might have a pretrained spm so they can use this one as a standard interface.

Reply:
Sure, but you can just write a factory function that returns an instance of https://github.com/pytorch/text/blob/084b0298e1c9886f5cbf40298bbda35a6841becc/torchtext/csrc/sentencepiece.cpp instead of this wrapper.

In fact, it already exists ```load_sp_model```. You can do ```load_sp_model(download(URL))``` and just have a list of URLs. The result of this will still be a JIT-able instance.